include/psa/crypto_se_driver.h

```diff
@@ -977,7 +977,21 @@ typedef psa_status_t (*psa_drv_se_generate_key_t)(psa_drv_se_context_t *drv_cont
  * If one of the functions is not implemented, it should be set to NULL.
  */
 typedef struct {
-    /** Function that allocates a slot. */
+    /** Function that allocates a slot for a key.
+     *
+     * The core calls this function to determine a slot number, then
```

Comment:
Why is this documentation here, not next to `psa_drv_se_allocate_key_t`?

Reply:
I'm not sure how to structure the documentation between the type and the function pointer. I think it makes more sense to document in the documentation of the type, which is where the parameters and return type are spelled out. But I don't think it makes sense to document the context in which the function is called there, and especially not information like “this function is called after that function” — from a type, it doesn't really make sense to reference a specific function rather than another type.

I'm open to suggestions to structure the documentation better.

Reply:
I would propose leaving just short notes here, like `Function that allocates a slot for a key.`, and provide all the documentation next to `psa_drv_se_allocate_key_t`. With the current structure, if someone wants to use this function, he/she has to read about it in both of these places, and that's confusing.